SU2_CFD/src/driver_structure.cpp

```diff
@@ -4396,49 +4508,49 @@ void CHBDriver::SetHarmonicBalance(unsigned short iZone) {
   for (iMGlevel = 0; iMGlevel <= config_container[ZONE_0]->GetnMGLevels(); iMGlevel++) {
 
     /*--- Loop over each node in the volume mesh ---*/
-    for (iPoint = 0; iPoint < geometry_container[ZONE_0][iMGlevel]->GetnPoint(); iPoint++) {
+    for (iPoint = 0; iPoint < geometry_container[ZONE_0][iInst][iMGlevel]->GetnPoint(); iPoint++) {
 
       for (iVar = 0; iVar < nVar; iVar++) {
         Source[iVar] = 0.0;
       }
 
       /*--- Step across the columns ---*/
-      for (jZone = 0; jZone < nZone; jZone++) {
+      for (jInst = 0; jInst < nInstHB; jInst++) {
 
         /*--- Retrieve solution at this node in current zone ---*/
         for (iVar = 0; iVar < nVar; iVar++) {
 
           if (!adjoint) {
-            U[iVar] = solver_container[jZone][iMGlevel][FLOW_SOL]->node[iPoint]->GetSolution(iVar);
-            Source[iVar] += U[iVar]*D[iZone][jZone];
+            U[iVar] = solver_container[ZONE_0][jInst][iMGlevel][FLOW_SOL]->node[iPoint]->GetSolution(iVar);
+            Source[iVar] += U[iVar]*D[iInst][jInst];
```

Comment:
How this would work for multi-zone problems? why is this now restricted to ZONE_0? In principle you need to calculate it for each geometrical zone. I am assuming that nTimeInstances is the number of time instances for each geometrical zone (otherwise the reg test would not pass).

Reply:
It would require just adding the loop on iZone and iInst for that. It is not restricted to ZONE_0; simply the loop was moved to jInst instead of jZone.